src/overlays/actors/ovl_Bg_Mori_Hashigo/z_bg_mori_hashigo.c

```diff
@@ -20,39 +40,256 @@ const ActorInit Bg_Mori_Hashigo_InitVars = {
+    s32 pad;
+    s32 localConst;
+    s32 pad2;
+
+    localConst = 0;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, &localConst);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, localConst);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_mori_hashigo.c", 164,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgMoriHashigo_InitCollider(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->dyna.actor, &sJntSphInit, this->colliderItems);
+
+    this->collider.list[0].dim.worldSphere.center.x = (s16)this->dyna.actor.posRot.pos.x;
+    this->collider.list[0].dim.worldSphere.center.y = (s16)this->dyna.actor.posRot.pos.y + 21;
+    this->collider.list[0].dim.worldSphere.center.z = (s16)this->dyna.actor.posRot.pos.z;
+    this->collider.list[0].dim.worldSphere.radius = 19;
+}
+
+s32 BgMoriHashigo_SpawnLadder(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    f32 sn;
+    f32 cs;
+    Vec3f pos;
+    Actor* ladder;
+
+    cs = Math_Coss(this->dyna.actor.shape.rot.y);
+    sn = Math_Sins(this->dyna.actor.shape.rot.y);
+
+    pos.x = 6.0f * sn + this->dyna.actor.posRot.pos.x;
+    pos.y = -210.0f + this->dyna.actor.posRot.pos.y;
+    pos.z = 6.0f * cs + this->dyna.actor.posRot.pos.z;
+
+    ladder = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_BG_MORI_HASHIGO, pos.x, pos.y,
+                                pos.z, this->dyna.actor.posRot.rot.x, this->dyna.actor.posRot.rot.y,
+                                this->dyna.actor.posRot.rot.z, 0);
+    if (ladder != NULL) {
+        return true;
+    } else {
+        // Ladder failure
+        osSyncPrintf("Error : 梯子の発生失敗(%s %d)(arg_data 0x%04x)\n", "../z_bg_mori_hashigo.c", 220,
+                     this->dyna.actor.params);
+        return false;
+    }
+}
+
+s32 BgMoriHashigo_InitClasp(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    static InitChainEntry sInitChain[] = {
+        ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_CONTINUE),
+        ICHAIN_U8(unk_1F, 3, ICHAIN_CONTINUE),
+        ICHAIN_F32(unk_4C, 40, ICHAIN_CONTINUE),
+        ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+    };
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.flags |= 1;
+    Actor_SetHeight(&this->dyna.actor, 55.0f);
+    BgMoriHashigo_InitCollider(this, globalCtx);
+    if ((this->dyna.actor.params == (s16)0xFFFF) && !BgMoriHashigo_SpawnLadder(this, globalCtx)) {
```

Comment:
Can -1 be used for these 0xFFFF values? If so we should go with that imo.
```suggestion
    if ((this->dyna.actor.params == -1) && !BgMoriHashigo_SpawnLadder(this, globalCtx)) {
```